src/catalog/catalog_entry/table_catalog_entry.cpp

```diff
@@ -236,12 +306,82 @@ unique_ptr<CatalogEntry> TableCatalogEntry::RenameColumn(ClientContext &context,
 	return make_unique<TableCatalogEntry>(catalog, schema, (BoundCreateTableInfo *)bound_create_info.get(), storage);
 }
 
+unique_ptr<CatalogEntry> TableCatalogEntry::AddGeneratedColumn(ClientContext &context, AddGeneratedColumnInfo &info) {
+	auto create_info = make_unique<CreateTableInfo>(schema->name, name);
+
+	// Copy all the columns
+	for (idx_t i = 0; i < columns.size(); i++) {
+		auto copy = columns[i].Copy();
+		create_info->columns.push_back(move(copy));
+	}
+
+	// now we copy all the generated columns
+	for (idx_t i = 0; i < generated_columns.size(); i++) {
+		auto copy = generated_columns[i].Copy();
+		create_info->generated_columns.push_back(move(copy));
+	}
+	// and we add the new generated column
+	info.new_column.oid = columns.size();
+	info.new_column.CheckValidity(create_info->columns, info.name);
+	create_info->generated_columns.push_back(info.new_column.Copy());
+
+	// Copy all the constraints
+	for (idx_t i = 0; i < constraints.size(); i++) {
+		auto constraint = constraints[i]->Copy();
+		create_info->constraints.push_back(move(constraint));
+	}
+
+	auto binder = Binder::CreateBinder(context);
+	auto bound_create_info = binder->BindCreateTableInfo(move(create_info));
+	auto result =
+	    make_unique<TableCatalogEntry>(catalog, schema, (BoundCreateTableInfo *)bound_create_info.get(), storage);
+
+	vector<string> names;
+	vector<LogicalType> types;
+
+	idx_t table_index = 0;
+
+	auto scan_function = TableScanFunction::GetFunction();
+	auto bind_data = make_unique<TableScanBindData>(result.get());
+
+	for (auto &col : result->columns) {
+		names.push_back(col.name);
+		types.push_back(col.type);
+	}
+
+	auto logical_get = make_unique<LogicalGet>(table_index, scan_function, move(bind_data), types, names);
+	binder->bind_context.AddBaseTable(table_index, result->name, names, types, vector<string>(), vector<LogicalType>(),
+	                                  *logical_get);
+
+	auto expr_binder = ExpressionBinder(*binder, context);
+	auto expression = info.new_column.expression->Copy();
+	// expr_binder.target_type = col.type;
+	auto bound_expression = expr_binder.Bind(expression);
+	if (!bound_expression) {
+		throw BinderException("Could not resolve the expression of generated column \"%s\"", info.new_column.name);
+	}
+	if (bound_expression->HasSubquery()) {
```

Comment:
This code seems to be mostly duplicated from the code we have in the create table, could we move this to the `Binder` and unify the code to avoid the duplication?